azure-kusto-data/azure/kusto/data/_cloud_settings.py

```diff
@@ -0,0 +1,82 @@
+from urllib.parse import urlparse
+
+
+class CloudInfo:
+    """ This class holds the data for a specific cloud instance. """
+
+    def __init__(self, auth_endpoint: str, kusto_client_app_id: str, redirect_uri: str):
+        self.aad_authority_uri = auth_endpoint
+        self.kusto_client_app_id = kusto_client_app_id
+        self.login_redirect_uri = redirect_uri  # will be used for interactive login
+
+
+class CloudSettings:
+    """ This class holds data for all cloud instances, and returns the specific data instance by parsing the dns suffix from a URL """
+
+    public_cloud_suffix = "windows.net"
+    moon_cake_cloud_suffix = "chinacloudapi.cn"
+    black_forest_cloud_suffix = "cloudapi.de"
+    fairfax_cloud_suffix = "usgovcloudapi.net"
+    us_nat_cloud_suffix = "core.eaglex.ic.gov"
+    us_sec_cloud_suffix = "core.microsoft.scloud"
+
+    _initialized = False
+    _cloud_info = {}
+    _suffix_list = [public_cloud_suffix, moon_cake_cloud_suffix, black_forest_cloud_suffix, fairfax_cloud_suffix, us_nat_cloud_suffix, us_sec_cloud_suffix]
+
+    @classmethod
+    def _init_once(cls):
+        if cls._initialized:
+            return
+
+        cls._initialized = True
+
+        cls._cloud_info[cls.public_cloud_suffix] = CloudInfo(
+            "https://login.microsoftonline.com", "db662dc1-0cfe-4e1c-a843-19a68e65be58", "https://microsoft/kustoclient"

```

Comment:
Are you going to remove those hardcoded settings? If not please use consts

Reply:
Remains to be seen.
Waiting for a reply from Azure AD if they can merge our client app ID into why.

Reply:
"https://microsoft/kustoclient"

Reply:
"db662dc1-0cfe-4e1c-a843-19a68e65be58"